app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java

```diff
@@ -1336,27 +1352,54 @@ public void onDialogDismissed(int dialogId) {
     public static class DatePickerFragment extends DialogFragment
             implements DatePickerDialog.OnDateSetListener {
 
-        final Context context;
-        final EditText textFieldEdit;
+        private static final String TEXT_FIELD_TO_EDIT_ARGUMENT_KEY = "text_field_to_edit";
+        private static final String CURRENT_DATE_ARGUMENT_KEY = "current_date";
+        private static final String MIN_DATE_ARGUMENT_KEY = "min_date";
+        private static final String MAX_DATE_ARGUMENT_KEY = "max_date";
+        private static final String PICK_DATE_REQUEST_KEY = "pick_date_request_key";
+        private static final String NEWLY_PICKED_DATE_ARGUMENT_KEY = "newly_picked_date";
+
+        LoyaltyCardField textFieldEdit;
         @Nullable
-        final Date minDate;
+        Date minDate;
         @Nullable
-        final Date maxDate;
+        Date maxDate;
+
+        public static DatePickerFragment newInstance(@NonNull LoyaltyCardField textField, @Nullable Date currentDate, @Nullable Date minDate, @Nullable Date maxDate) {
+            Bundle args = new Bundle();
+            args.putSerializable(TEXT_FIELD_TO_EDIT_ARGUMENT_KEY, textField);
+            args.putSerializable(CURRENT_DATE_ARGUMENT_KEY, currentDate);
+            args.putSerializable(MIN_DATE_ARGUMENT_KEY, minDate);
+            args.putSerializable(MAX_DATE_ARGUMENT_KEY, maxDate);
+            DatePickerFragment fragment = new DatePickerFragment();
+            fragment.setArguments(args);
+            return fragment;
+        }
+
+        public interface OnDatePickListener {
```

Comment:
I would suggest having nested classes/interfaces at the top of their containing class, not in between its methods.